src/jit/optimizer.cpp

```diff
@@ -5711,34 +5711,76 @@ bool Compiler::optNarrowTree(GenTree* tree, var_types srct, var_types dstt, Valu
         switch (tree->gtOper)
         {
             case GT_AND:
+                noway_assert(genActualType(tree->gtType) == genActualType(op1->gtType));
                 noway_assert(genActualType(tree->gtType) == genActualType(op2->gtType));
 
-                // Is op2 a small constant than can be narrowed into dstt?
-                // if so the result of the GT_AND will also fit into 'dstt' and can be narrowed
-                if ((op2->gtOper == GT_CNS_INT) && optNarrowTree(op2, srct, dstt, NoVNPair, false))
+                GenTree* opToNarrow;
+                opToNarrow = nullptr;
+                GenTree** otherOpPtr;
+                otherOpPtr = nullptr;
+                bool canNotNarrowOperand;
```

Comment:
Could you change this to `canNarrowOperand`? Boolean values with negative names are generally confusing.

Reply:
I agree that in general it's preferrable not to have boolean values with negative names; however, I think in this case canNarrowOperand would be even more confusing. When the existing canNotNarrowOperand boolean is true, it means "we know at least one of the operands can't be narrowed". When the existing canNotNarrowOperand boolean is false, it means "we haven't found an operand that can't be narrowed" but it doesn't mean that "we found an operand that can be narrowed".

Reply:
The purpose of this boolean is to prevent calling optNarrowTree(doit=false) on each operand twice: once in the GT_AND case and once in COMMON_BINOP case. If we happen to know that one of the operands can't be narrowed, there is no point going to COMMON_BINOP case.

Reply:
I don't think that necessitates a negative boolean. Why can't you simply reverse the sense of each use of this? You would still check whether `canNarrowOperand` is true before falling through to the `COMMON_BINOP` case.

Reply:
Getting out my bikeshed paintbrush ....

How about `foundOperandThatBlocksNarrowing` ?

Reply:
I like that. @CarolEidt I'll rename to foundOperandThatBlocksNarrowing if you are ok with that.

Reply:
I can live with it, but from a pure logic standpoint I don't understand how it is ever the case that replacing "canNotDoX" with "canDoX" and reversing all references results in different behavior - or is there some obvious thing I am missing?

Reply:
I guess the confusion here is that In this case the reverse of canNotDoX is not canDoX but eitherCanDoXOrDon'tKnowWhetherCanDoX. That's why I'm reluctant to use canDoX in this case.

Reply:
I'll rename to foundOperandThatBlocksNarrowing, it makes the code more clear than canNotNarrowOperand or canNarrowOperand.